runtime/test/manifest-test.js

```diff
@@ -727,6 +727,41 @@ ${particleStr1}
     assert.deepEqual(['aa', 'hello'], slotConn.tags);
     assert.lengthOf(Object.keys(slotConn.providedSlots), 1);
   });
+  it('SLANDLES recipe slots with tags', async () => {
+    let manifest = await Manifest.parse(`
+      particle SomeParticle in 'some-particle.js'
+        \`consume Slot slotA #aaa
+          \`provide Slot slotB #bbb
+        recipe
+          \`slot 'slot-id0' #aa #aaa as s0
+          SomeParticle
+            slotA consume s0 #aa #hello
+            slotB provide
+    `);
+    // verify particle spec
+    assert.lengthOf(manifest.particles, 1);
+    let spec = manifest.particles[0];
+    assert.lengthOf(spec.connections, 2);
+    let slotSpec = spec.connections[0];
+    assert.deepEqual(slotSpec.tags, ['aaa']);
+    assert.lengthOf(slotSpec.dependentConnections, 1);
+    let providedSlotSpec = slotSpec.dependentConnections[0];
+    assert.deepEqual(providedSlotSpec.tags, ['bbb']);
+
+    // verify recipe slots
+    assert.lengthOf(manifest.recipes, 1);
+    let recipe = manifest.recipes[0];
+    assert.lengthOf(recipe.handles, 1);
+    let recipeSlot = recipe.handles.find(s => s.id == 'slot-id0');
+    assert(recipeSlot);
+    assert.deepEqual(recipeSlot.tags, ['aa', 'aaa']);
+
+    let slotConn = recipe.particles[0].connections['slotA'];
+    assert(slotConn);
+    assert.deepEqual(['aa', 'hello'], slotConn.tags);
+    //TODO(jopra): Give recipes the dependentConnections syntax+handling
```

Comment:
..maybe.. :)

We should look at the recipe API when we get there and decide whether it makes sense for each given use to distinguish between slots and handles.

There are some where it definitely will, and e.g. from a particle developer perspective you want to keep the 2 concepts separate. But for strategizing, etc., maybe we can ignore the difference.

Reply:
Okay. It will be good to revisit it then.
I'm interested in adding the functionality, but will need some time and guidance on it.